diff --git a/pex/environment.py b/pex/environment.py
index b8e9201..3df1f0f 100644
--- a/pex/environment.py
+++ b/pex/environment.py
@@ -19,6 +19,7 @@ from pkg_resources import (
 
 from .common import die, open_zip, safe_mkdir, safe_rmtree
 from .interpreter import PythonInterpreter
+from .orderedset import OrderedSet
 from .package import distribution_compatible
 from .pex_builder import PEXBuilder
 from .pex_info import PexInfo
@@ -185,6 +186,7 @@ class PEXEnvironment(Environment):
 
     working_set = WorkingSet([])
     resolved = self._resolve(working_set, all_reqs)
+    sys_path_backup = sys.path[:]
 
     for dist in resolved:
       with TRACER.timed('Activating %s' % dist, V=2):
@@ -196,4 +198,9 @@ class PEXEnvironment(Environment):
 
         dist.activate()
 
+    if self._pex_info.inherit_path:
+        # prioritize import of distributions contained within this pex
+        sys_path_additions = list(OrderedSet(sys.path) - OrderedSet(sys_path_backup))
+        sys.path = sys_path_additions + sys_path_backup
+
     return working_set
